### PR TITLE
Add Subgraph and tasks

### PR DESCRIPTION
**Pull request by JamesLefrere**
- Add a subgraph to track MerkleDrops
- Define a `MerkleDropTranches` contract to manage the subgraph in a fairly cost-efficient way
- Fix hardhat tasks and add tasks for tranches

**TODO**

- [x] Fix CI (looks like codegen needed)
- [x] Ropsten subgraph deploy (for testing frontend)
- [x] Validate with frontend

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,15 +6,44 @@ A lightweight Merkle Drop contract.
 
 This project uses Hardhat.
```
Comment (rafaelugolini) [inline README.md:8]: missing here how to get started:

```
yarn typechain
yarn compile
```

File: README.md [hunk 2]
```diff
@@ -6,15 +6,44 @@ A lightweight Merkle Drop contract.
...
+
+### Registering a MerkleDrop on MerkleDropTranches (useful for subgraphs)
+
+    yarn hardhat registerMerkleDrop --merkle-drop-tranches 0x... --merkle-drop 0x...
+
+### Adding a new tranche URI (useful for subgraphs)
+
+    yarn hardhat addTrancheURI --merkle-drop-tranches 0x... --merkle-drop 0x... --id 0  --balances ./tranche0.json
+
+## Development
```
Comment (rafaelugolini) [inline README.md]: I would add here some of the steps to compile the project locally

```
yarn typechain
yarn compile
```

Testing could be part of `## Development` and `### Installation in your project` should a new section.

What do you think?

Comment (JamesLefrere) [reply README.md]: I'll add it 👍

**Review by rafaelugolini — APPROVED** [review]
Approving since we test together

**Pull request merged**